doc/tutorials/imgproc/shapedescriptors/bounding_rects_circles/bounding_rects_circles.markdown

```diff
@@ -22,6 +22,97 @@ This tutorial code's is shown lines below. You can also download it from
 Explanation
 -----------
 
+The main function is rather simple, as follows from the comments we
+-#  Open the image, convert it into grayscale and blur it to get rid of the noise
+    @code{.cpp}
+    /// Load source image
+    src = imread( argv[1], 1 );
+
+    /// Convert image to gray and blur it
+    cvtColor( src, src_gray, CV_BGR2GRAY );
+    blur( src_gray, src_gray, Size(3,3) );
+    @endcode
+-# Create a window with header "Source" and display the source file in it.
+    @code{.cpp}
+    /// Create Window
+    char* source_window = "Source";
+    namedWindow( source_window, CV_WINDOW_AUTOSIZE );
+    imshow( source_window, src );
+    @endcode
+-# Create a trackbar on the source_window and assign a callback function to it
+   In general callback functions are used to react to some kind of signal, in our
+   case it's trackbar's state change.
+    @code{.cpp}
+    createTrackbar( " Threshold:", "Source", &thresh, max_thresh, thresh_callback );
+    @endcode
+
+The callback function `threshold` does all the interesting job.
+-# Writes to `threshold_output` the threshold of the grayscale picture, (you can check out about thresholding [here](http://docs.opencv.org/2.4/doc/tutorials/imgproc/threshold/threshold.html))
+    @code{.cpp}
+    /// Detect edges using Threshold
+    threshold( src_gray, threshold_output, thresh, 255, THRESH_BINARY );
+    @endcode
+-# Finds contours and saves them to the vectors `contour` and `hierarchy` ([more about findContours](http://docs.opencv.org/2.4/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html#void%20findContours(InputOutputArray%20image,%20OutputArrayOfArrays%20contours,%20OutputArray%20hierarchy,%20int%20mode,%20int%20method,%20Point%20offset))).
```

Comment:
You should use either simple `cv::findContours` either `@ref cv::findContours` (if first does not work) to create a link to the existing function documentation.